api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -608,7 +608,8 @@ type AKSExtension struct {
 	ExtensionType *string `json:"extensionType"`
 
 	// Plan is the plan of the extension.
-	Plan *ExtensionPlan `json:"plan"`
+	// +optional
+	Plan *ExtensionPlan `json:"plan,omitempty"`
```

Comment:
Since CAPZ is going to mark `Plan` as optional, should CAPZ be opinionated and default to a plan?

Reply:
Some extensions actually don't allow you to specify a plan, such as `microsoft.Flux`. Plan will have to be null for those extensions to work.